x-pack/legacy/plugins/ml/public/data_frame/pages/data_frame_new_pivot/components/wizard/wizard.tsx

```diff
@@ -0,0 +1,174 @@
+
+import { isKibanaContext, getCreateRequestBody, KibanaContext } from '../../../../common';
+
+import {
+  StepDefineExposedState,
+  StepDefineForm,
+  StepDefineSummary,
+  getDefaultStepDefineState,
+} from '../step_define';
+import { getDefaultStepCreateState, StepCreateForm, StepCreateSummary } from '../step_create';
+import { getDefaultStepDetailsState, StepDetailsForm, StepDetailsSummary } from '../step_details';
+import { WizardNav } from '../wizard_nav';
+
+enum WIZARD_STEPS {
+  DEFINE,
+  DETAILS,
+  CREATE,
+}
+
+interface DefinePivotStepProps {
+  isCurrentStep: boolean;
+  stepDefineState: StepDefineExposedState;
+  setCurrentStep: React.Dispatch<React.SetStateAction<WIZARD_STEPS>>;
+  setStepDefineState: React.Dispatch<React.SetStateAction<StepDefineExposedState>>;
+}
+
+const StepDefine: SFC<DefinePivotStepProps> = ({
+  isCurrentStep,
+  stepDefineState,
+  setCurrentStep,
+  setStepDefineState,
+}) => {
+  const definePivotRef = useRef(null);
+
+  return (
+    <Fragment>
+      <div ref={definePivotRef} />
+      {isCurrentStep && (
+        <Fragment>
+          <StepDefineForm onChange={setStepDefineState} overrides={{ ...stepDefineState }} />
+          <WizardNav
+            next={() => setCurrentStep(WIZARD_STEPS.DETAILS)}
+            nextActive={stepDefineState.valid}
+          />
+        </Fragment>
+      )}
+      {!isCurrentStep && <StepDefineSummary {...stepDefineState} />}
+    </Fragment>
+  );
+};
+
+export const Wizard: SFC = React.memo(() => {
+  const kibanaContext = useContext(KibanaContext);
+
+  if (!isKibanaContext(kibanaContext)) {
+    return null;
+  }
+
+  const indexPattern = kibanaContext.currentIndexPattern;
+
+  // The current WIZARD_STEP
+  const [currentStep, setCurrentStep] = useState(WIZARD_STEPS.DEFINE);
+
+  // The DEFINE state
+  const [stepDefineState, setStepDefineState] = useState(getDefaultStepDefineState(kibanaContext));
+
+  // The DETAILS state
+  const [stepDetailsState, setStepDetailsState] = useState(getDefaultStepDetailsState());
+
+  const stepDetails =
+    currentStep === WIZARD_STEPS.DETAILS ? (
+      <StepDetailsForm onChange={setStepDetailsState} overrides={stepDetailsState} />
+    ) : (
+      <StepDetailsSummary {...stepDetailsState} />
+    );
+
+  const transformConfig = getCreateRequestBody(
+    indexPattern.title,
+    stepDefineState,
+    stepDetailsState
+  );
+
+  // The CREATE state
+  const [stepCreateState, setStepCreateState] = useState(getDefaultStepCreateState);
+
+  const stepCreate =
+    currentStep === WIZARD_STEPS.CREATE ? (
+      <StepCreateForm
+        createIndexPattern={stepDetailsState.createIndexPattern}
+        transformId={stepDetailsState.transformId}
+        transformConfig={transformConfig}
+        onChange={setStepCreateState}
+        overrides={stepCreateState}
+      />
+    ) : (
+      <StepCreateSummary />
+    );
+
+  // scroll to the currently selected wizard step
+  /*
+  function scrollToRef() {
```

Comment:
Is this going to be added in eventually?

Reply:
yes, I'd like to bring that back at some point :)